src/groups/bmq/bmqimp/bmqimp_brokersession.cpp

```diff
@@ -5239,7 +5245,13 @@ BrokerSession::createConfigureQueueContext(const bsl::shared_ptr<Queue>& queue,
 
             bmqp_ctrlmsg::Subscription subscription(d_allocator_p);
 
-            subscription.sId() = cit->first.id();
+            static bsls::AtomicUint s_nextId(0);
```

Comment:
This static local variable initialization is not thread safe in general. The probability to observe a bug is small though.

Here are some details and advices:

https://bloomberg.github.io/bde-resources/doxygen/bde_api_prod/group__bslmt__once.html#_details

And here is how we did a safe initialization for another component:

https://github.com/bloomberg/blazingmq/blob/08a6d16b1b73d1d71113f80f7c22fcdd3e5392d8/src/groups/bmq/bmqt/bmqt_correlationid.cpp#L33-L42